integration-test/test-connectors/src/test/java/org/wso2/ei/test/connector/soap/SoapConnectorTest.java

```diff
@@ -0,0 +1,129 @@
+
+/**
+ * Integration tests for soap connector.
+ */
+public class SoapConnectorTest {
+    private static final Logger log = LoggerFactory.getLogger(SoapConnectorTest.class);
+
+    private boolean serverStarted;
+
+    @BeforeClass
+    @Description("Starts the server with the SoapConnectorTest.bal file")
+    public void startServer() {
+        serverStarted = TestUtils.startServer(
+                "integration-test/test-connectors/src/test/java/org/wso2/ei/test/connector/soap/SoapConnectorTest.bal");
+    }
+
+    @Test
+    @Description("Tests the sendReceive method of the soap connector")
+    public void testSendReceive() {
+        String symbol = "ABC";
+        String responseBody = sendToBackend(symbol, 200, "sendReceive");
+
+        String responseSymbol = responseBody.split("<ax21:symbol>")[1].split("</ax21:symbol>")[0].trim();
+        Assert.assertEquals(responseSymbol, symbol);
+    }
+
+    @Test
+    @Description("Tests the sendReboust method")
+    public void testSendRobust() {
+        String symbol = "DEF";
+        String responseBody = sendToBackend(symbol, 200, "sendRobust");
+
+        Assert.assertEquals(responseBody, "Success");
+    }
+
+    @Test
+    @Description("Tests the error returned of the sendRobust method of the soap connector")
+    public void testSendRobustTimeout() {
+        String symbol = "ABC";
+        String responseBody = sendToBackend(symbol, 504, "sendRobustTimeout");
+
+        Assert.assertEquals(responseBody, "Gateway Timeout");
+    }
+
+    @Test
+    @Description("Tests the fireAndForget method of the soap connector")
+    public void testFireAndForget() {
+        String symbol = "GHI";
+        sendToBackend(symbol, 200, "fireAndForget");
+    }
+
+    /**
+     * This is a convenient method for sending to the backend and asserts the status.
+     *
+     * @param symbol         the symbol to send to backend.
+     * @param expectedStatus the expected status from the call to the backend.
+     * @param invoke         the path to invoke when calling the backend.
+     * @return the response as a string.
+     */
+    private String sendToBackend(String symbol, int expectedStatus, String invoke) {
+        if (!serverStarted) {
+            Assert.fail("Error running the test, server is not started");
+        }
+
+        String requestPayload = "<ser:getSimpleQuote xmlns:ser=\"http://services.samples\">\n"
+                + "<ser:symbol>" + symbol + "</ser:symbol>\n"
+                + "</ser:getSimpleQuote>";
+
+        String url = "http://localhost:9090/SimpleStockQuoteService/" + invoke;
+
+        HttpClient client = new HttpClient();
+        PostMethod post = new PostMethod(url);
+
+        try {
+            RequestEntity requestEntity = new StringRequestEntity(requestPayload, "application/xml", "UTF-8");
+
+            post.setRequestEntity(requestEntity);
+
+            int status = client.executeMethod(post);
+            Assert.assertEquals(status, expectedStatus);
```

Comment:
Isn't it better to add assert messages using the overloaded method, to be logged when the assert fails?